src/lib/sqlite/global.ts

```diff
@@ -1,17 +1,44 @@
-import { createQueryExecutor, createSQLiteAPI } from './query-executor'
-import type { QueryExecutor } from './query-executor'
-import { drizzle } from 'drizzle-orm/sqlite-proxy'
+import {
+  createLocalDb,
+  createLocalFoldersStore,
+  createLocalSnippetsStore,
+  createQueryExecutor,
+  createSQLiteAPIV2
+} from './wa-sqlite'
+import { derived } from 'svelte/store'
+import { globalStateStore, settingsStore } from '$lib/utitlities/global'
+import { createRemoteFoldersStore, createRemoteSnippetStore, createSqlitergExecutor } from '$lib/sqlite/sqliterg'
+import {
+  createHigherFoldersStore,
+  createHigherSnippetsStore,
+  createSnippetsDataManager,
+  createSnippetsDataStateStore, reloadRemoteFoldersStore
+} from '$lib/utitlities/synchronization'
+import { waitUntil } from '$lib/utitlities/wait-until'
 
-export async function createLocalDb(executor: QueryExecutor) {
-  return drizzle(async (queryString, params, method) => {
-    const result = await executor.execute(queryString, ...params)
-    if (method === 'get' && result.length > 0) {
-      return {rows: result[0]}
-    }
-    return {rows: result}
-  })
-}
-
-export const sqlite3 = await createSQLiteAPI()
+export const sqlite3 = await createSQLiteAPIV2()
 export const executor = await createQueryExecutor(sqlite3, 'crypta')
-export const localDb = await createLocalDb(executor)
+export const localDb = createLocalDb(executor)
+export const sqlitergExecutorStore = derived(
+  settingsStore,
+  (settings) => {
+    return createSqlitergExecutor(
+      settings.serverURL,
+      settings.username,
+      settings.password,
+    )
+  }
+)
+export const localSnippetsStore = await createLocalSnippetsStore(executor, globalStateStore)
+export const remoteSnippetsStore = await createRemoteSnippetStore(globalStateStore, sqlitergExecutorStore)
+await waitUntil(remoteSnippetsStore.isAvailable, 100, 1000)
```

Comment:
If the stored URL is unreachable/has error, then this would block the application for 1s, which is quite bad.